server/src/main/java/org/opensearch/index/translog/transfer/TransferService.java

```diff
@@ -54,7 +54,8 @@ void uploadBlobs(
         Set<TransferFileSnapshot> fileSnapshots,
         final Map<Long, BlobPath> blobPaths,
         ActionListener<TransferFileSnapshot> listener,
-        WritePriority writePriority
+        WritePriority writePriority,
+        final Map<TransferFileSnapshot, InputStream> transferFileMetadata
```

Comment:
Interface looks inconsistent. I would prefer TransferFileSnapshot to hold the metadata InputStream, indicating if the file snapshot has a corresponding metadata